test/intl402/NumberFormat/prototype/formatRange/argument-to-Intlmathematicalvalue-throws.js

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2021 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-Intl.NumberFormat-formatRange
+description: >
+  "formatRange" basic tests when argument cannot be converted using ToIntlMathematicalValue
+info: |
+  Intl.NumberFormat.prototype.formatRange( start, end )
+  (...)
+  4. Let x be ? ToIntlMathematicalValue(start).
+  5. Let y be ? ToIntlMathematicalValue(end).
+locale: [en-US]
+features: [Intl.NumberFormat-formatRange]
+---*/
+
+
+const nf = new Intl.NumberFormat("en-US", {signDisplay: "exceptZero"});
+
+// Throw if arguments cannot be cast using the method ToIntlMathematicalValue
```

Comment:
This is more strict than the spec requires.  According to the spec, `formatRange` throws a `RangeError` if the number is NaN.  The only case where it should throw a `TypeError` is if the argument is illegal in either `ToPrimitive` or `ToNumber`.

https://tc39.es/proposal-intl-numberformat-v3/out/numberformat/proposed.html#sec-tointlmathematicalvalue

Note that ToIntlMathematicalValue is supposed to have the same semantics as ECMA-262 ToNumeric, except for the part about retaining additional precision.